test/signalrclient-e2e-tests/connection_tests.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include "stdafx.h"
+#include <string>
+#include "cpprest\details\basic_types.h"
+#include "cpprest\json.h" 
+#include "connection.h"
+#include "hub_connection.h"
+#include "url.h"
+
+TEST(connection_tests, connection_status_start_stop)
+{
+    auto conn = std::make_shared<signalr::connection>(url + U("raw-connection"));
+    auto test_event = std::make_shared<pplx::event>();
+
+    try
+    {
+        conn->start().get();
+        ASSERT_EQ(conn->get_connection_state(), signalr::connection_state::connected);
+        conn->stop().get();
+        ASSERT_EQ(conn->get_connection_state(), signalr::connection_state::disconnected);
+        conn->start().get();
+        ASSERT_EQ(conn->get_connection_state(), signalr::connection_state::connected);
+        test_event->set();
+    }
+    catch (...)
+    {
+        ASSERT_TRUE(false);
+    }
+
+    ASSERT_FALSE(test_event->wait(15000));
+}
+
+TEST(connection_tests, send_message)
+{
+    auto conn = std::make_shared<signalr::connection>(url + U("raw-connection"));
+    auto message = std::make_shared<utility::string_t>();
+    auto test_event = std::make_shared<pplx::event>();
+
+    conn->set_message_received([message, test_event](const utility::string_t& payload)
+    {
+        *message = payload;
+        test_event->set();
+    });
+
+    try
+    {
+        conn->start().then([conn]()
+        {
+            web::json::value obj;
+            obj[U("type")] = web::json::value::number(0);
+            obj[U("value")] = web::json::value::string(U("test"));
+            return conn->send(obj.serialize());
+
+        }).get();
+    }
+    catch (...)
+    {
+        ASSERT_TRUE(false);
```

Comment:
I think I should have meant `cout` because `.what()` is `std::string`/`char *`. However then I realized it would be even better to remove `try...catch` blocks if you are blocking with `.get()` because we don't expect exceptions and if they happen gtest will take care of it - it will fail the test and will write the exception message for you.